src/typeInference/annotateAST.ts

```diff
@@ -287,21 +287,38 @@ export function annotate(ast: IAST, staticContext: StaticContext): SequenceType
 			return annotateUnaryLookup(ast, ncName);
 
 		// TypeSwitch
-		case 'typeSwitchExpr':
-			const arg = annotate(astHelper.getFirstChild(ast, 'argExpr') as IAST, staticContext);
-			const clauses = astHelper
+		case 'typeswitchExpr':
+			const argumentType = annotate(
+				astHelper.getFirstChild(ast, 'argExpr')[1] as IAST,
+				staticContext
+			);
+			const caseClausesReturns = astHelper
 				.getChildren(ast, 'typeswitchExprCaseClause')
-				.map((a) => annotate(a, staticContext));
-			const defaultCase = annotate(
-				astHelper.getFirstChild(ast, 'typeSwitchExprDefaultClause') as IAST,
+				.map((a) =>
+					annotate(astHelper.followPath(a, ['resultExpr'])[1] as IAST, staticContext)
+				);
+			const defaultCaseReturn = annotate(
+				astHelper.followPath(ast, ['typeswitchExprDefaultClause', 'resultExpr'])[1] as IAST,
 				staticContext
 			);
-			return annotateTypeSwitchOperator(ast);
+			return annotateTypeSwitchOperator(
+				ast,
+				argumentType,
+				caseClausesReturns,
+				defaultCaseReturn
+			);
 
 		case 'quantifiedExpr':
 			astHelper.getChildren(ast, '*').map((a) => annotate(a, staticContext));
 			return annotateQuantifiedExpr(ast);
 
+		// XQuery nodes
+		case 'x:stackTrace':
+			return annotate(astHelper.getChildren(ast, '*')[2], staticContext);
+
+		case 'resultExpr':
```

Comment:
I don't think this one should be placed here since its not really a separate AST node. Its more a property of the typeswitchExpr.

Reply:
Indeed, I was testing around and forgot to remove them, good catch.